src/components/AddTodoForm/AddTodoForm.tsx

```diff
@@ -0,0 +1,182 @@
+import { useCallback, useState } from 'react';
+import { User } from '../../types/User';
+import './index.scss';
+
+enum ChangeType {
+  TITLE,
+  USER,
+  COMPLETED,
+}
+
+type Props = {
+  handleSubmit: (
+    user: number,
+    title: string,
+    completed: boolean,
+  ) => void;
+  users: User[],
+};
+
+export const AddTodoForm = ({ handleSubmit, users }: Props) => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState(0);
+  const [completed, setCompleted] = useState(false);
+  const [errors, setErrors] = useState({
+    isUserNotSelected: false,
+    isTitleEmpty: false,
+  });
+
+  const clearForm = useCallback(
+    () => {
+      setTitle('');
+      setUser(0);
+      setCompleted(false);
+    },
+    [],
+  );
+
+  const handleChange = useCallback(
+    (
+      e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+      field: ChangeType,
+    ) => {
+      switch (field) {
+        case ChangeType.TITLE:
+          setTitle(e.target.value);
+          setErrors(prev => ({
+            isUserNotSelected: prev.isUserNotSelected,
+            isTitleEmpty: false,
+          }));
+          break;
+        case ChangeType.USER:
+          setUser(Number.parseInt(e.target.value, 10));
+          setErrors(prev => ({
+            isUserNotSelected: false,
+            isTitleEmpty: prev.isTitleEmpty,
+          }));
+          break;
+        default: // ChangeType.COMPLETED
+          setCompleted(prev => !prev);
+          break;
+      }
+    }, [],
+  );
+
+  const onSubmit = (
+    e: React.SyntheticEvent | React.MouseEvent<HTMLButtonElement>,
+  ) => {
+    e.preventDefault();
+    setErrors({
+      isUserNotSelected: user === 0,
+      isTitleEmpty: title.length === 0,
+    });
+
+    if (user === 0 || title.length === 0) {
```

Comment:
user === 0 seems really illogical. I expect user to be an object - we get number instead